File: analysis/private_ledger.py
"""Resolve private research inputs without committing workstation or media names.

The ledger is deliberately external to Git. Missing configuration fails closed;
an alias must never silently become a relative output path on the system disk.
"""
import json
import os
from pathlib import Path


def private_value(index: str) -> str:
    platform = "WINDOWS" if os.name == "nt" else "POSIX"
    location = os.environ.get(f"VOLLEYCUT_PRIVATE_LEDGER_{platform}") or os.environ.get("VOLLEYCUT_PRIVATE_LEDGER")
    if not location:
        raise RuntimeError("Set VOLLEYCUT_PRIVATE_LEDGER to the private research ledger")
    ledger = json.loads(Path(location).read_text(encoding="utf-8"))
    value = ledger.get("privateValues", {}).get(index)
    if not isinstance(value, str) or not value:
        raise ValueError(f"Missing private ledger index: {index}")
    return value

File: lib/server/private-ledger.mjs
import { readFileSync } from "node:fs";

// Server/research tooling only. Never import this module into a client component.
export function privateValue(index) {
  const platform = process.platform === "win32" ? "WINDOWS" : "POSIX";
  const location = process.env[`VOLLEYCUT_PRIVATE_LEDGER_${platform}`] || process.env.VOLLEYCUT_PRIVATE_LEDGER;
  if (!location) throw new Error("Set VOLLEYCUT_PRIVATE_LEDGER to the private research ledger");
  const ledger = JSON.parse(readFileSync(location, "utf8"));
  const value = ledger.privateValues?.[index];
  if (typeof value !== "string" || !value) throw new Error(`Missing private ledger index: ${index}`);
  return value;
}
